src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java

```diff
@@ -202,8 +208,9 @@ private void requestMaterialsState()
     @Override
     protected IAIState waitForRequests()
     {
-        if (job.hasWorkOrder() && building.getNeededResources().isEmpty())
+        if (job.hasWorkOrder() && building.getNeededResources().isEmpty() && !recalculated)
         {
+            recalculated = true;
```

Comment:
Shouldn't it also set this to false when it completes the job or something?

Reply:
I think so yes

Reply:
I think so yes

Reply:
No, this is specifically if the builder is unloaded mid build. They have a request open and then don't recalc the list, if the builder is unloaded again, the variable auto resets to false